velox/exec/Unnest.cpp

```diff
@@ -217,40 +307,55 @@ const Unnest::UnnestChannelEncoding Unnest::generateEncodingForChannel(
       }
     }
   }
+
+  if (range.size > 1) {
+    firstLastRowGenerator(range.start + range.size - 1, 0, range.lastRowEnd);
+  }
+
   return {elementIndices, nulls, identityMapping};
 }
 
 VectorPtr Unnest::generateOrdinalityVector(
-    vector_size_t start,
-    vector_size_t size,
+    const RowRange& range,
     vector_size_t numElements) {
   auto ordinalityVector =
       BaseVector::create<FlatVector<int64_t>>(BIGINT(), numElements, pool());
 
   // Set the ordinality at each result row to be the index of the element in
   // the original array (or map) plus one.
   auto* rawOrdinality = ordinalityVector->mutableRawValues();
-  for (auto row = start; row < start + size; ++row) {
+
+  VELOX_DCHECK_GT(range.size, 0);
+  const auto firstRowEnd =
+      range.size == 1 ? range.lastRowEnd : rawMaxSizes_[range.start];
+  const auto maxSize = firstRowEnd - firstRowStart_;
+  std::iota(rawOrdinality, rawOrdinality + maxSize, firstRowStart_ + 1);
```

Comment:
Would it make sense to add rowRange.forEachRow(<lambda>) method to simplify this and similar code?

```
rowRange.forEachRow([&](auto rowStart, auto rowSize) {
    std::iota(rawOrdinality, rawOrdinality + rowSize, rowStart);
    rawOrdinality += rowSize;
});
```